earthpy/tests/test_mask.py

```diff
@@ -28,16 +28,26 @@ def test_masked_arr_returned():
     masked = make_apply_mask(im, im_mask, [0,4])
     assert np.ma.is_masked(masked)
 
-# def test_val_list_provided:
-#     """Tuple works just fine. what about some other object type? like? """
-#
-#     with pytest.raises(AttributeError):
-#         make_apply_mask(im,
-#                     mask_arr=im_mask,
-#                     vals=[0,4])
+def test_val_list_provided():
+    """Tuple works just fine. what about some other object type? like?
+        how about a singular value? if not that, maybe a dictionary?"""
+
+    with pytest.raises(AttributeError):
+        make_apply_mask(im,
+                    mask_arr=im_mask,
+                    vals=0)
 
 # Test that a mask is provided to apply mask that doesn't have 1's in it.
 # should return - there's nothing to mask here!
+def test_mask_provided_with_no_ones():
+
+    im_mask_no_ones = im_mask*0
+    with pytest.raises(ValueError):
+        make_apply_mask(im, im_mask_no_ones, [3,4])
 
 # test that a user provides an array with a mask already. does it merge properly?
-# and return a masked array with the values expected?
\ No newline at end of file
+# and return a masked array with the values expected?
+def test_im_with_mask_as_masked_array():
```

Comment:
i might be missing something here so please say the word if i am... where is the actual test in this function @joemcglinchy  ? i don't see any sort of assert or raises statement here but again might be missing something.

Reply:
yes, because the section of code that was not previously hit does not raise an error... it only checks if there is not a single value 1 element in the mask array.

Reply:
I wasn't sure how to test for this one... we could try an assert statement to make sure the output is of type `np.ma.MaskedArray`... would that suffice?

Reply:
totally @joemcglinchy  

here is an example using hasattr
https://github.com/earthlab/earthpy/blob/master/earthpy/tests/test_clip.py#L67

I think you can do something similar using
assert `np.ma.is_masked()`

https://docs.scipy.org/doc/numpy-1.15.1/reference/generated/numpy.ma.is_masked.html

does that seem reasonable to you?

Reply:
added to it in the most recent commit. push away!

Reply:
woo hoo!! thank you @joemcglinchy  i'll merge this now.